src/harness/externalCompileRunner.ts

```diff
@@ -36,32 +39,88 @@ abstract class ExternalCompileRunnerBase extends RunnerBase {
                     if (install.status !== 0) throw new Error(`NPM Install for ${directoryName} failed!`);
                 }
                 Harness.Baseline.runBaseline(`${this.kind()}/${directoryName}.log`, () => {
-                    const result = cp.spawnSync(`node`, [path.join(__dirname, "tsc.js")], { cwd, timeout, shell: true });
-                    // tslint:disable-next-line:no-null-keyword
-                    return result.status === 0 && !result.stdout.length && !result.stderr.length ? null : `Exit Code: ${result.status}
-Standard output:
-${result.stdout.toString().replace(/\r\n/g, "\n")}
-
-
-Standard error:
-${result.stderr.toString().replace(/\r\n/g, "\n")}`;
+                    return this.report(cp.spawnSync(`node`, [path.join(__dirname, "tsc.js")], { cwd, timeout, shell: true }), cwd);
                 });
             });
         });
     }
 }
 
 class UserCodeRunner extends ExternalCompileRunnerBase {
-    public readonly testDir = "tests/cases/user/";
-    public kind(): TestRunnerKind {
+    readonly testDir = "tests/cases/user/";
+    kind(): TestRunnerKind {
         return "user";
     }
+    report(result: any) {
+        // tslint:disable-next-line:no-null-keyword
+        return result.status === 0 && !result.stdout.length && !result.stderr.length ? null : `Exit Code: ${result.status}
+Standard output:
+${result.stdout.toString().replace(/\r\n/g, "\n")}
+
+
+Standard error:
+${result.stderr.toString().replace(/\r\n/g, "\n")}`;
+    }
 }
 
 class DefinitelyTypedRunner extends ExternalCompileRunnerBase {
-    public readonly testDir = "../DefinitelyTyped/types/";
-    public workingDirectory = this.testDir;
-    public kind(): TestRunnerKind {
+    readonly testDir = "../DefinitelyTyped/types/";
+    workingDirectory = this.testDir;
+    kind(): TestRunnerKind {
         return "dt";
     }
+    report(result: any, cwd: string) {
+        const stdout = filterExpectedErrors(result.stdout.toString(), cwd)
+        const stderr = result.stderr.toString()
+        // tslint:disable-next-line:no-null-keyword
+        return !stdout.length && !stderr.length ? null : `Exit Code: ${result.status}
+Standard output:
+${stdout.replace(/\r\n/g, "\n")}
+
+
+Standard error:
+${stderr.replace(/\r\n/g, "\n")}`;
+    }
+}
+
+function filterExpectedErrors(errors: string, cwd: string): string {
+    return breaks(errors.split("\n"), s => /^\w+/.test(s)).filter(isExpectedError(cwd)).map(lines => lines.join("\n")).join("\n");
+}
+function isExpectedError(cwd: string) {
+    return (error: string[]) => {
+        if (error.length === 0) {
+            return true;
+        }
+        const match = error[0].match(/(.+\.ts)\((\d+),\d+\): error TS/);
+        if (!match) {
+            return true;
+        }
+        const errlines = fs.readFileSync(path.join(cwd, match[1]), { encoding: "utf8" }).split("\n");

```

Comment:
Nit: Use an array destructure on the `match` immediately after the `if` so I don't have to remember what the nth capture group refers to.
Also, I think this variable is misnamed since it's the entire text of the file. Not just the errors.